### DISCUSSION TITLE
Some Problems with VSCode Extension

### DISCUSSION THREAD

**Discussion by cemalgnlts** [volarjs/volar.js#146] [Q&A] [answered]
Hi,

I tried to make this example myself: https://github.com/volarjs/vscode-typescript-web

However, I get the following errors:

```js
import {
  createConnection,
  startLanguageServer, // Module '"@volar/language-server/browser"' has no exported member 'startLanguageServer'.
  LanguageServerPlugin, // '"@volar/language-server/browser"' has no exported member named 'LanguageServerPlugin'. Did you mean 'LanguagePlugin'?
  InitializationOptions
} from "@volar/language-server/browser";

import createTsService from "volar-service-typescript";

config.services ??= {};
config.services.typescript = createTsService(); // This expression is not callable.
  // Type 'typeof import("/home/cemalgnlts/jside/node_modules/volar-service-typescript/index")' has no call signatures.
```

I wonder if the APIs have changed. What their current usage has been replaced with.

**Comment by johnsoncodehk** [comment, marked as answer]
Yes, the API has changed. You can refer to this migration PR. Since the Vue language tools it depends on have not been upgraded to Volar v2, and the @volar/cdn used for ATA has been removed, it cannot be upgraded to Volar v2 for the time being.

https://github.com/volarjs/vscode-typescript-web/pull/6

**Reply by cemalgnlts** [reply]
Great, thanks.
